src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -63,12 +63,136 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
             DiagnosticBag diagnostics)
         {
             var type = (NamedTypeSymbol)this.BindType(node.Type, diagnostics);
+
+            var operators = type.GetMembers(WellKnownMemberNames.IsOperatorName);
```

Comment:
to me it feels reasonable to look in base if there is no suitable match in the derived type. We do this for other operators.